packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -113,42 +113,44 @@ class TreeItem extends UIComponent<WithAsProp<TreeItemProps>> {
       e.preventDefault()
       e.stopPropagation()
 
-      _.invoke(this.props, 'onTitleClick', e, this.props)
+      this.handleTitleClick(e)
     },
-    collapseOrReceiveFocus: e => {
-      const { items, open } = this.props
+    receiveFocus: e => {
+      e.preventDefault()
+      e.stopPropagation()
 
+      this.itemRef.current.focus()
+    },
+    collapse: e => {
       e.preventDefault()
+      e.stopPropagation()
 
-      // Focuses the title if the event comes from a child item.
-      if (e.currentTarget !== e.target && items && items.length) {
-        e.stopPropagation()
-        this.itemRef.current.focus()
-      } else if (open) {
-        e.stopPropagation()
-        _.invoke(this.props, 'onTitleClick', e, this.props)
-      }
+      this.handleTitleClick(e)
     },
-    expandOrPassFocus: e => {
-      const { open } = this.props
+    expand: e => {
+      e.preventDefault()
+      e.stopPropagation()
 
+      this.handleTitleClick(e)
+    },
+    passFocus: e => {
```

Comment:
Suggestion: `focusSubtree`?  It would be nice to say _where_ we are passing the focus to.  Currently, it says to pass it, but as a dev implementing this I'm not sure where to pass it to.